variables.tf

```diff
@@ -177,170 +200,372 @@ variable "additional_node_groups" {
   default = {}
 }
 
-variable "cidr" {
-  type        = string
-  default     = "10.0.0.0/16"
-  description = "The IPv4 CIDR block for the VPC."
-  validation {
-    condition = (
-      try(cidrhost(var.cidr, 0), null) == regex("^(.*)/", var.cidr)[0] &&
-      try(cidrnetmask(var.cidr), null) == "255.255.0.0"
-    )
-    error_message = "Argument base_cidr_block must be a valid CIDR block."
-  }
-}
-
-variable "pod_cidr" {
-  type        = string
-  default     = "100.64.0.0/16"
-  description = "The IPv4 CIDR block for the VPC."
-  validation {
-    condition = (
-      try(cidrhost(var.pod_cidr, 0), null) == regex("^(.*)/", var.pod_cidr)[0] &&
-      try(cidrnetmask(var.pod_cidr), null) == "255.255.0.0"
-    )
-    error_message = "Argument base_cidr_block must be a valid CIDR block."
-  }
-}
-
-variable "use_pod_cidr" {
-  type        = bool
-  description = "Use additional pod CIDR range (ie 100.64.0.0/16) for pod/service networking"
-  default     = true
-}
-
-variable "eks_master_role_names" {
-  type        = list(string)
-  description = "IAM role names to be added as masters in eks."
-  default     = []
-}
-
-variable "vpc_id" {
-  type        = string
-  description = "Optional VPC ID, it will bypass creation of such, public_subnets and private_subnets are also required."
-  default     = null
-}
-
-variable "public_subnets" {
-  type        = list(string)
-  description = "Optional list of public subnet ids"
-  default     = null
-}
+variable "network" {
+  description = <<EOF
+    vpc = {
+      id = Existing vpc id, it will bypass creation by this module.
+      subnets = {
+        private = Existing private subnets.
+        public  = Existing public subnets.
+        pod     = Existing pod subnets.
+      }), {})
+    }), {})
+    network_bits = {
+      public  = Number of network bits to allocate to the public subnet. i.e /27 -> 32 IPs.
+      private = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+      pod     = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+    }
+    cidrs = {
+      vpc     = The IPv4 CIDR block for the VPC.
+      public  = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+      private = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+      pod     = The IPv4 CIDR block for the Pod subnets.
+    }
+    use_pod_cidr = Use additional pod CIDR range (ie 100.64.0.0/16) for pod/service networking.
+  EOF
```

Comment:
network bits and cidrs have the same description so something is wrong.
why is use_pod_cidr outside?

Reply:
>network bits and cidrs have the same description.
Bad paste.
>why is use_pod_cidr outside?
to facilitate the cidrs validation,  https://github.com/dominodatalab/terraform-aws-eks/blob/427b677d340da19b41ba65bd67046cba04f85f35/submodules/network/variables.tf#L76